src/components/TextField/TextField.tsx

```diff
@@ -151,35 +130,35 @@ export interface Props {
  * import {TextField} from "@chanzuckerberg/eds";
  */
 export const TextField = ({
+  'aria-describedby': ariaDescribedBy,
   className,
-  type = 'text',
-  id,
-  required,
-  label,
   disabled,
   fieldNote,
-  fieldButtonText,
-  fieldButtonScreenReaderText,
-  fieldButtonOnClick,
+  id,
   inputWithin,
-  name,
-  value,
-  defaultValue,
-  onChange,
-  placeholder,
-  readOnly,
-  iconName,
-  inverted,
-  hideLabel,
-  maxLength,
-  inputMode,
-  'aria-describedby': ariaDescribedBy,
   isError,
-  optionalLabel,
-  requiredLabel,
-  title,
+  label,
+  optionalIndicator,
+  required,
+  type = 'text',
   ...other
 }: Props) => {
+  if (process.env.NODE_ENV !== 'production' && !label && !other['aria-label']) {
+    throw new Error('You must provide a visible label or aria-label');
+  }
+
+  const shouldRenderOverline = label || required || optionalIndicator;
+  const indicator = required
+    ? 'Required'
+    : optionalIndicator
+    ? 'Optional'
+    : null;
```

Comment:
Small thing but it looks like if you move the `shouldRenderOverline` down under `indicator` you could check for `label || indicator` since `indicator` checks for `required` and `optionalIndicator`.

```
  const indicator = required
    ? 'Required'
    : optionalIndicator
    ? 'Optional'
    : null;
  const shouldRenderOverline = label || indicator;
```
I'm also wondering if you should make `shouldRenderOverline` a boolean like
```
  const shouldRenderOverline = !!(label || indicator);
```
because it looks like right now it's a string?

Reply:
Ohhh great suggestions, will adopt